### DISCUSSION TITLE
Does filtering need a delimiter?

### DISCUSSION THREAD

**Discussion by cbhiii** [ESPresense/ESPresense#312] [Q&A]
When looking to filter what is sent (or exclude) via MQTT, am I supposed to use full IDs?
Can I use wildcards?
Do multiple IDs need to be comma delimited?

I've not seen this documented anywhere. I appreciate any advice to properly list them so I don't cause any software issues. Thanks.

Which syntax is correct in the examples for the "Include" (If set will only...) field listed below:
```
apple:1005:9-24, apple:1005:9-26
  or 
apple:1005:9-24 apple:1005:9-26
  or
apple:1005:*
```

**Comment by DTTerastar** [comment]
It is space delimited, and globs aren't supported, just prefixes.

**Reply by AlfredJKwack** [reply]
Hi @DTTerastar  

What would be the approach when you have associated your iPhone(s) with the ESPresense base sation (initially to obtain the `irk:XXX`). At this stage I see the device's actual name in the ID's column of ESPresense's fingerprints.

Putting the `irk:XXX` items in the filters seems not to work and kills _any_ updates from being sent. Since device names can actually have spaces they also can't be entered.

![Question1](https://user-images.githubusercontent.com/557102/189742900-f031bfe5-7b2e-48e3-a8dc-19c65148cc53.png)
The sceenshot above is from ESPresense Version: v3.0.4

**Reply by DTTerastar** [reply]
Under settings you can publish a new config that has a aliased id with no space.  I do need to make the ui slugify whatever a person puts in there.

**Reply by AlfredJKwack** [reply]
@DTTerastar. Thanks for the quick reply. Could you be a little more explicit? Not quite sure which part of the settings allow a user to set a device name (slugified) for an arbitrary irk/id. If I get it I'd be happy to update the doc with a PR accordingly as well to help clarify it for the rest of the community.

**Comment by DTTerastar** [comment]
You can publish (retained) to `espresense/settings/irk:{irk}` with a body of:

{"id":"irk:darrells-m1-air"}

Also supported are name and rss@1m.  I wouldn't bother documenting other than here since I plan on making the gui do it for you.

**Reply by DTTerastar** [reply]
You can also rename something publish to `espresense/settings/name:thingy` body:

{ "id": "nf-dev", name: "My thing-a-ling", "rssi@1m":-64 }

**Comment by mterry63** [comment]
@DTTerastar can you give an example of an alias or rename command for an iBeacon device? If I try to set the publish filter with a list of 4 iBeacons, nothing is sent (it appears the string is too long and everything is filtered). If I set the publish filter to iBeacon it appears only the 1st discovered iBeacon is sent. Can I use the alias/rename to allow a filter that will be able to list 4 short names with spaces delimiting?

**Reply by mterry63** [reply]
Nevermind, I see that each receiver now publishes its own beacon settings and it's a good example of how to publish your own. Using that I was able to create shorter id names and create a filter that worked well.
